lib/msf/core/handler/reverse_http.rb

```diff
@@ -215,17 +215,25 @@ def lookup_proxy_settings
   #
   def on_request(cli, req, obj)
     resp = Rex::Proto::Http::Response.new
+    info = process_uri_resource(req.relative_resource)
+    uuid = info[:uuid] || Msf::Payload::UUID.new
 
-    print_status("#{cli.peerhost}:#{cli.peerport} Request received for #{req.relative_resource}...")
+    # Configure the UUID architecture and payload if necessary
+    uuid.arch      = obj.arch if uuid.arch.nil?
+    uuid.platform  = obj.platform if uuid.platform.nil?
```

Comment:
Is the usual approach to use `||=`?
``` ruby
uuid.arch ||= obj.arch
uuid.platform ||= obj.platform
```

Seems to be quite common:
```
$ grep -r "||=" . | wc -l
1824
```
To be clear, I don't really mind :)